dmoj/judgeenv.py

```diff
@@ -177,44 +180,54 @@ def load_env(cli=False, testsuite=False):  # pragma: no cover
     try:
         with open(model_file) as init_file:
             env.update(yaml.safe_load(init_file))
-
-            if getattr(args, 'judge_name', None):
-                env['id'] = args.judge_name
-
-            if getattr(args, 'judge_key', None):
-                env['key'] = args.judge_key
-
-            problem_dirs = env.problem_storage_root
-            if problem_dirs is None:
-                if not testsuite:
-                    raise SystemExit(
-                        'problem_storage_root not specified in "%s"; no problems available to grade' % model_file
-                    )
     except IOError:
         if not is_docker:
             raise
 
-    # Populate cache and send warnings
-    get_problem_roots(warnings=True)
+    if getattr(args, 'judge_name', None):
+        env['id'] = args.judge_name
 
-    def get_path(x, y):
-        return utf8text(os.path.normpath(os.path.join(x, y)))
+    if getattr(args, 'judge_key', None):
+        env['key'] = args.judge_key
 
-    if isinstance(problem_dirs, str):
-        problem_dirs = [problem_dirs]
+    if env.problem_storage_globs:
+        problem_globs = env.problem_storage_globs
+        # Populate cache and send warnings
+        get_problem_roots(warnings=True)
 
-    problem_watches = []
-    for dir in problem_dirs:
-        if isinstance(dir, ConfigNode):
-            for _, recursive_root in dir.iteritems():
-                problem_watches.append(get_path(_root, recursive_root))
-        else:
-            problem_watches.append(get_path(_root, dir))
+        if isinstance(problem_globs, str):
+            problem_globs = [problem_globs]
+
+        problem_watches = problem_globs
+
+    elif env.problem_storage_root:
+        startup_warnings.append('problem_storage_root is deprecated, use problem_storage_globs instead')
+        problem_dirs = env.problem_storage_root
+        # Populate cache and send warnings
+        get_problem_roots(warnings=True)
+
+        def get_path(x, y):
+            return utf8text(os.path.normpath(os.path.join(x, y)))
+
+        if isinstance(problem_dirs, str):
+            problem_dirs = [problem_dirs]
+
+        problem_watches = []
+        for dir in problem_dirs:
+            if isinstance(dir, ConfigNode):
+                for _, recursive_root in dir.iteritems():
+                    problem_watches.append(get_path(_root, recursive_root))
+            else:
+                problem_watches.append(get_path(_root, dir))
+
+    if problem_globs is None and problem_dirs is None:
+        if not testsuite:
+            raise SystemExit('problem_storage_globs not specified in "%s"; no problems available to grade' % model_file)
 
     if testsuite:
         if not os.path.isdir(args.tests_dir):
             raise SystemExit('Invalid tests directory')
-        problem_dirs = [args.tests_dir]
+        problem_globs = [os.path.join(args.tests_dir, '*')]
```

Comment:
Something we could do (when we're ready to replace `problem_storage_root` with `problem_storage_globs`) is to replace the test regexes with test globs

Reply:
This sounds like a fine idea to leave as a comment :)